File: AmirImamTask.AngularUI/src/app/models/Transaction.ts
import { Guid } from "guid-typescript";
import { TransactionDetail } from "./TransactionDetail";

export class Transaction {
    public Id: string = Guid.EMPTY;
    public TransactionDate: Date = new Date();
    public StoreId?: string;
    public ItemsCount: number = 0;
    public TotalPrices: number = 0;
    public Net: number = 0;

    public TransactionDetails?: TransactionDetail[];
}

File: AmirImamTask.Entities/Entities/Transaction.cs
namespace AmirImamTask.Entities;
[Table(nameof(Transaction))]
public class Transaction : EntityBase
{
    public DateTime TransactionDate { get; set; }
    public Guid? StoreId { get; set; }
    public Store? Store { get; set; }

    public int ItemsCount { get; set; }
    public double TotalPrices { get; set; }
    public double Net { get; set; }

    public ICollection<TransactionDetail>? TransactionDetails { get; set; }
}
